src/index.ts

```
@@ -315,3 +420,106 @@ export function generateSalt(byteCount = 32): string {
+ *
+ * @param exportedKey - The object to check.
+ * @returns Whether or not the object is an `ExportedEncryptionKey`.
+ */
+export function isExportedEncryptionKey(
+  exportedKey: unknown,
+): exportedKey is ExportedEncryptionKey {
+  return (
+    exportedKey !== null &&
+    typeof exportedKey === 'object' &&
+    'key' in exportedKey &&
+    'derivationOptions' in exportedKey &&
+    isKeyDerivationOptions(exportedKey.derivationOptions)
+  );
+}
+
+/**
+ * Returns the `CryptoKey` from the provided encryption key.
+ * If the provided key is a `CryptoKey`, it is returned as is.
+ *
+ * @param encryptionKey - The key to unwrap.
+ * @returns The `CryptoKey` from the provided encryption key.
+ */
+export function unwrapKey(encryptionKey: EncryptionKey | CryptoKey): CryptoKey {
+  return isEncryptionKey(encryptionKey) ? encryptionKey.key : encryptionKey;
+}
+
+/**
+ * Checks if the provided vault is an updated encryption format.
+ *
+ * @param vault - The vault to check.
+ * @returns Whether or not the vault is an updated encryption format.
+ */
+export function isVaultUpdated(vault: string): boolean {
+  const { keyMetadata } = JSON.parse(vault);
+  return (
+    isKeyDerivationOptions(keyMetadata) &&
+    keyMetadata.algorithm === DEFAULT_DERIVATION_PARAMS.algorithm &&
+    keyMetadata.params.iterations ===
+      DEFAULT_DERIVATION_PARAMS.params.iterations
+  );
+}
+
+/**
+ * Updates the provided vault, re-encrypting
+ * data with a safer algorithm if one is available.
+ *
+ * If the provided vault is already using the latest available encryption method,
+ * it is returned as is.
+ *
+ * @param vault - The vault to update.
+ * @param password - The password to use for encryption.
+ * @returns A promise resolving to the updated vault.
+ */
+export async function updateVault(
+  vault: string,
+  password: string,
+): Promise<string> {
+  if (isVaultUpdated(vault)) {
+    return vault;
+  }
+
+  return encrypt(password, await decrypt(password, vault));
+}
```

This function can be used by the extension and other clients to migrate their vaults. 

As this PR is non-breaking, this is not mandatory, as the library would silently use the new method when encrypting the vault again. But with this helper, the library gets the responsibility of knowing whether the vault is using the latest safe configuration available, resulting in a no-op if the vault is already up-to-date